### PR TITLE
fix: Modify entry logic to use Typer, colorize output, add interactive prompts

### PR DESCRIPTION
**Pull request by mbeacom**
## Summary

The goal of this PR is to swap out argparse with the more robust third-party module: `typer`, providing shell completion support, full `click` support, and much more.
Additionally, this PR enabled a confirmation prompt when executing eksupgrade following the pre-flight checks (requiring user input of y/N) and colorizes all outputs based on disposition of the operation originating the output.

Resolves: #31 

### Changes

Swaps out the logger usage throughout most of the project with `typer.secho` pre-canned variants in `eksupgrade.utils@echo_*`, adds the `--interactive`/`--no-interactive` flags to disable prompts, and updates the entry script to use typer.

### User experience

Colorized outputs based on status, shell completion support, clearer help text, and interactive/non-interactive workflows.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

- [x] I have performed a self-review of this change
- [x] Changes have been tested
- [ ] Changes are documented

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: eksupgrade/src/preflight_module.py [hunk 1]
```diff
@@ -338,13 +331,13 @@ def addon_version(
                     "env": daemon_set_item.spec.template.spec.containers[0].env,
                 }
                 version = version_str.split(".")
-                logger.info("VPC CNI found version: %s - version_str: %s", version, version_str)
-                logger.info("Likely desired vpc-cni version: %s", cni_target_version)
+                echo_info(f"VPC CNI found version: {version} - version_str: {version_str}")
```
Comment (quixoticmonk) [inline eksupgrade/src/preflight_module.py:334]: - We should drop the `version` from the log as it is a list which can be confusing ( like in the discussion [here](https://github.com/aws-samples/eks-cluster-upgrade/discussions/85#discussioncomment-5362567)) 

`echo_info(f"VPC CNI found version: {version_str}")`

Comment (mbeacom) [reply eksupgrade/src/preflight_module.py]: Updated.

**Review by bryantbiggs — APPROVED** [review]
Looks great, ship it!

**Review by quixoticmonk — APPROVED** [review]
This is awesome.Good to merge

**Pull request merged**